src/gui/Src/Gui/CPUStack.cpp

```diff
@@ -406,7 +424,80 @@ QString CPUStack::paintContent(QPainter* painter, dsint rowBase, int rowOffset,
         painter->drawText(QRect(x + 4, y , w - 4 , h), Qt::AlignVCenter | Qt::AlignLeft, makeAddrText(wVa));
         return QString();
     }
-    return HexDump::paintContent(painter, rowBase, rowOffset, col, x, y, w, h);;
+    else if(col == 1) // paint stack data
+    {
+        // get call stack
+        DBGCALLSTACK callstack;
+        memset(&callstack, 0, sizeof(DBGCALLSTACK));
+        // TODO: need to cache the call stack data.
+        DbgFunctions()->GetCallStack(&callstack);

```

Comment:
I think the performance loss of getting the callstack on every repaint will be very noticeable, especially with the unpredictable performance of dbghelp. Possibly this needs to be integrated in the dbg where the callstack is retrieved from a thread. If I have time tonight I will check if this is easy to do and do it myself.